faster_whisper/transcribe.py

```diff
@@ -218,77 +198,61 @@ def get_language_and_tokenizer(self, audio, task=None, language=None):
         return language, language_probability, task
     
     def audio_split(self, audio, segments, sampling_rate):
+        "Returns splitted audio chunks as iterator"
         for seg in segments:
             f1 = int(seg['start'] * sampling_rate)
             f2 = int(seg['end'] * sampling_rate)
             yield {'inputs': audio[f1:f2]}
 
-    def transcribe_stream(
-        self, audio: Union[str, np.ndarray], vad_segments, batch_size=None, num_workers=0, language=None, task=None, log_progress = False, combined_progress=False
-    ) -> TranscriptionResult:
+    def transcribe(
+        self, audio: Union[str, np.ndarray], vad_segments: List[dict], batch_size=16, num_workers=0, language=None, task=None, log_progress = False
+    ) ->Tuple[Iterable[BatchedSegment], dict]: 
+        """transcribe audio in chunks specified in vad_segments and return BatchedSegment generator with language info.
+
+        args:
+            audio: complete audio file as numpy array/path to the audio file for batched transcription.
+            vad_segments: list of dictionaries each containing "start" and "end" keys, specifying the start and end voiced regions of audio chunks.
+            batch_size: the maximum number of parallel requests to model for decoding.
+            num_workers: to enable true parallelism when running the model, same as the transcribe function argument in WhisperModel class.
+            language: The language spoken in the audio. It should be a language code such as "en" or "fr".
+            task: either "transcribe" or "translate".
+            log_progress: whether to show progress bar or not. 
+        """
+        sampling_rate = self.model.feature_extractor.sampling_rate
+
         if isinstance(audio, str):
             audio = decode_audio(audio)
 
+        #if no segment split is provided, consider a single vad segment from start to end. No benefit in this case.
+        if not vad_segments: 
```

Comment:
If no VAD segments is an option then it should be documented and vad_segments  should be optional then.

Reply:
It was not an option before, rather a required parameter. For efficient working of the batched whisper, you need non empty vad_segments. If empty/None comes we should gracefully still run it. 

Or do you think adding the vad model inside and handling it inside is better?

Reply:
Added optional and docstring for vad_segments